learningequality/studio
# Add functionality for exporting published public channels to kolibri_public models

## Summary
### Description of the change(s) you made
* Adds a ChannelMapper class to map from kolibri_content models to kolibri_public models
* Draws inspiration from the Studio ContentNode duplication functionality to try to do this efficiently within the confines of the Django ORM
* Adds a tree id manager model for kolibri_public to prevent concurrency errors for creating trees as a precaution
* Adds tests of the ChannelMapper class, first for a simple import, and then for a repeated import
* Updates the publishing code to run this after a published channel database has been successfully uploaded
* Adds a management command to backfill kolibri_public for already published public channels

### Manual verification steps performed
1. I mostly relied on the automated tests
2. Before pulling in this PR (or by not restarting the services) publish a channel and make it public, but without the new code in publishing here that would automatically export it, then run the management command: `./contentcuration/manage.py export_channels_to_kolibri_public` - confirm that your recently published, public channel is imported

Fixes https://github.com/learningequality/studio/issues/3954

Caveat - I used the same ordering that we use for the V1 endpoint to set the ordering that is used for the V2 endpoint - we should update this at some point, but this seemed the wrong time to do so.
